roomserver/internal/input/input_events.go

```diff
@@ -128,9 +128,9 @@ func (r *Inputer) processRoomEvent(
 	if roomInfo == nil && !isCreateEvent {
 		return fmt.Errorf("room %s does not exist for event %s", event.RoomID(), event.EventID())
 	}
-	_, senderDomain, err := gomatrixserverlib.SplitID('@', event.Sender())
+	sender, err := event.UserID()
 	if err != nil {
-		return fmt.Errorf("event has invalid sender %q", input.Event.Sender())
+		return fmt.Errorf("event has invalid sender %q", event.SenderID())
```

Comment:
Confusing if this ever happens as the error was caused by calling `.UserID` not `.SenderID` and we drop `err` to the floor.